### ISSUE TITLE
Where clause does not take effect if conditions are false with "allow filtering" enabled

### ISSUE THREAD

**Issue by disney007** [scylladb/scylla#6764]
This is Scylla's bug tracker, to be used for reporting bugs only.
If you have a question about Scylla, and not a bug, please ask it in
our mailing-list at scylladb-dev@googlegroups.com or in our slack channel.

- [x] I have read the disclaimer above, and I am reporting a suspected malfunction in Scylla.

*Installation details*
Scylla version (or git commit hash):  scylladb/scylla:3.2.4
Cluster size: 500M
OS (RHEL/CentOS/Ubuntu/AWS AMI): MAC

## Create table and insert data
```
create keyspace example
with replication = {
'class' : 'SimpleStrategy',
'replication_factor' : 1
};

create table if not exists example.table_a (   
    name    	        text,   
    category              text,
    ms		        timestamp,
    value                   decimal,   
    primary key((name), category, ms)
);

insert into example.table_a(name, category, ms, value)
values('NAME1', 'ABC', '2020-05-03 17:12:13.000+0000', 12.1);

insert into example.table_a(name, category, ms, value)
values('NAME1', 'ABC', '2020-05-03 18:12:13.000+0000', 10.1);
```
## query with "allow filtering" enabled
```
select name, category, sum(value) from example.table_a 
where ms > '2020-05-03 17:16:13.000+0000' and ms < '2020-05-03 17:15:13.000+0000'
group by name, category 
allow filtering;
```
Actual Result:
```
 name  | category | system.sum(value)
-------+----------+-------------------
 NAME1 |      ABC |              22.2
```
The sum value is wrong, the execpted value should be `0.0` because the condition`ms > '2020-05-03 17:16:13.000+0000' and ms < '2020-05-03 17:15:13.000+0000'` is always false

**Comment by nagarajbhat-ppro** [comment]
The above issue is faced on docker image with version 4.0 as well. We hit this issue on scylladb-cloud running opensource version 4.

**Comment by dekimir** [comment]
This is a duplicate of #5799

**Comment by dekimir** [comment]
Should be fixed when #5763 is merged.

**Comment by avikivity** [comment]
#5763 is not a bag of bug fixes. Having a rewrite fix things unexpectedly is a bad sign. We'll also need to backport the fix, and #5763 is impossible to backport.

**Comment by dekimir** [comment]
I didn't mean to say #5763 is a bag of fixes, just that when #5763 lands (as it already has, once), this issue will go away.  Yes, it's a bad sign, but the whole point of the rewrite is that this area is replete with assorted bad signs.

Backports will require a fundamentally different approach than master; we'll have to untangle the old single_column_restriction::slice, bounds_ranges, and do_filter.  Ideally, we would do it for all the bugs marked as fixed by 6862727.

**Comment by avikivity** [comment]
Closing as duplicate of #5799
